syft/frameworks/torch/tensors/interpreters/additive_shared.py

```diff
@@ -566,6 +594,24 @@ def chunk(tensor_shares, *args, **kwargs):
 
         module.chunk = chunk
 
+        @overloaded.function
+        def roll(tensor_shares, shifts, **kwargs):
+            results = {}
+
+            for worker, share in tensor_shares.items():
+                if isinstance(shifts, dict):
+                    results[worker] = torch.roll(share, shifts[worker].get().item(), **kwargs)
```

Comment:
the local client (so P2 in the paper implem) learns here shifts[worker].get().item() which is `i - g mod n` right? Can P2 learn `g` then, and then `r` and then deduce `ind_n`?

Reply:
I didn't follow you. `shifts[worker].get().item()` is `-r % n` (see 4) in the max pool_deriv function).
Thinking again about it, I think it is not secure to have P2 learning the shift because it could retrieve the output of the function...

Reply:
Oh, I remember now!
If we consider the max pool_deriv protocol, `share` and `shifts[worker]` are PointerTensors here, so there should be a way not to get them. But as `shifts[worker]` is a tensor and the roll function takes an int here, I need to use the .item() method which raises an error when called instead of get() on a PointerTensor.
It works if I remove this error raising and if I use shifts[worker].item() but as I don't know why the error was there in the first place, I don't want to break anything